parser.y

```diff
@@ -162,11 +165,27 @@ expr: ID { $$ = std::make_unique<IdExprNode>($1); }
   /* equality 6.5.9 */
   | expr EQ expr { $$ = std::make_unique<EqualToExprNode>($1, $3); }
   | expr NE expr { $$ = std::make_unique<NotEqualToExprNode>($1, $3); }
-  | '(' expr ')' { $$ = $2; }
   /* assignment 6.5.16 */
   | ID '=' expr { $$ = std::make_unique<SimpleAssignmentExprNode>($1, $3); }
   ;
 
+unary_expr: postfix_expr { $$ = $1; }
+  | PP unary_expr { $$ = std::make_unique<IncrExprNode>($2); }
+  | MM unary_expr { $$ = std::make_unique<DecrExprNode>($2); }
+  | '-' unary_expr { $$ = std::make_unique<NegExprNode>($2); }
+  | '!' unary_expr { $$ = std::make_unique<NotExprNode>($2); }
+  /* TODO: implement pointer type */
+  | '&' unary_expr { $$ = std::make_unique<AddrExprNode>($2); }
+  | '*' unary_expr { $$ = std::make_unique<DereferExprNode>($2); }
+  /* TODO: implement bitwise operations */
+  | '~' unary_expr { $$ = std::make_unique<BitCompExprNode>($2); }
+  ;
+
+postfix_expr: ID { $$ = std::make_unique<IdExprNode>($1); }
```

Comment:
I suggest renaming the symbol. I believe it would be suitable to introduce `primary_expr` for these expressions, as they appear to be unrelated to _postfix_ and _prefix_. They serve as the minimum unit of expression.

I agree that the reference manual seems to have designed these zig-zag grammars in a rather intricate manner. :disappointed_relieved: